lib/java_buildpack/framework/instana_agent.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2024 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module JavaBuildpack
+  module Framework
+    # Encapsulates the functionality for enabling Instana support.
+    class InstanaAgent < JavaBuildpack::Component::VersionedDependencyComponent
+
+      # Creates an instance
+      #
+      # @param [Hash] context a collection of utilities used the component
+      def initialize(context)
+        @application = context[:application]
+        @component_name = self.class.to_s.space_case
+        @configuration = context[:configuration]
+        @droplet = context[:droplet]
+
+        @version, @uri = standalone_agent_download_url if supports?
+        @logger = JavaBuildpack::Logging::LoggerFactory.instance.get_logger InstanaAgent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download_jar
```

Comment:
**question**: where is this `download_jar` defined? Or is this a general download capability that uses our  `@version` and `@uri`?

Reply:
download_jar is capability provided by java buildpack.
That is defined in the 
https://github.com/cloudfoundry/java-buildpack/blob/1ca7a11b757146a2d36617cc0431915355bd1e4b/lib/java_buildpack/component/versioned_dependency_component.rb#L69 

https://github.com/cloudfoundry/java-buildpack/blob/1ca7a11b757146a2d36617cc0431915355bd1e4b/lib/java_buildpack/component/base_component.rb#L90

It is general download capability which uses version and ur provided by us.